gridstatus/isone.py

```diff
@@ -227,6 +227,122 @@ def get_load_forecast(self, date, end=None, verbose=False):
 
         return df
 
+    def get_solar_forecast(self, date, end=None, verbose=False):
+        """Return solar forecast published on a specific date
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-solar-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Solar",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    def get_wind_forecast(self, date, end=None, verbose=False):
+        """Return wind forecast published on a specific date
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-wind-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Wind",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def _get_solar_or_wind_forecast(
+        self,
+        date,
+        end=None,
+        resource_type="Wind",
+        verbose=False,
+    ):
+        """Return solar or wind forecast published on a specific date
+
+        Resource type can be "Solar" or "Wind"
+        """
+        if date == "latest":
+            date = pd.Timestamp.now(tz=self.default_timezone)
+
+        value_name = f"{resource_type.capitalize()} Forecast"
+
+        url = (
+            f"https://www.iso-ne.com/transform/csv/wphf?start={date.strftime('%Y%m%d')}"
+        )
+
+        df = _make_request(url, skiprows=[0, 1, 2, 3, 5], verbose=verbose)
+
+        df.columns = df.iloc[0]
+        df = df.drop(columns=["D", "Date"], index=[0]).reset_index(drop=True)
+
+        data = df.melt(
+            id_vars=["Hour Ending"],
+            var_name="Date",
+            value_name=value_name,
+        ).dropna(subset=[value_name])
+
+        import IPython
+
+        IPython.core.interactiveshell.InteractiveShell.ast_node_interactivity = (
+            "last_expr_or_assign"
+        )
+        IPython.embed()
+
+        # for DST end transitions isone uses 02X to represent repeated 1am hour
```

Comment:
`02x` is same way it is handled in LMP data. does it make sense to converge the logic? 

https://github.com/WillKoehrsen/gridstatus/blob/548e3561978407d6009c32007e927dd2da36f4c1/gridstatus/isone.py#L521-L522

Reply:
Yep, I refactored to consolidate the logic in one function.